toolchain/check/type_completion.cpp

```diff
@@ -252,14 +252,14 @@ auto TypeCompleter::AddNestedIncompleteTypes(SemIR::Inst type_inst) -> bool {
     }
     case CARBON_KIND(SemIR::StructType inst): {
       for (auto field : context_->struct_type_fields().Get(inst.fields_id)) {
-        Push(field.type_id);
+        Push(context_->types().GetTypeIdForTypeInstId(field.type_inst_id));
```

Comment:
Do we expect to have all of these calls to `GetTypeIdForTypeInstId` long term, or is there a plan to use type ids in fewer places so they are needed lesss?

Reply:
I've been thinking about introducing a `TypeInstId` and using that in more places where we used to use `TypeId` and now are starting to use `InstId`. Looks like Dana got there first though (#5288)!